### ISSUE TITLE
AWS request payer

### ISSUE THREAD

**Issue by TiaanThera** [sentinel-hub/sentinelhub-py#43]
After 6 August AWS has implemented the requester pays to l1c bucket. Is there a way to still access sentinel l1c data by just adding the header of RequestPayer='requester' to a normal tile request as was done when requesting L2A data. If there is, in what python class would it be added and where. Sorry i am still new to using sentinel hub and python. At the moment access denied is shown when quering l1c but not l2a as  i am when config option is set to true request contructer is called that doesn't include requester paid

**Comment by AleksMat** [comment]
Hi TiaanThera,

since version `2.1.0` the package supports download from Requester Pays bucket. Please check these [configuration instructions](http://sentinelhub-py.readthedocs.io/en/latest/configure.html#amazon-s3-capabilities) for details.

Basically you have to switch from http to s3 bucket in package's configuration file. But we soon plan to release a new version which will access Requester Pays bucket by default.

**Comment by TiaanThera** [comment]
Thank you for the quick reply. Yes the function to set the bucket to use or the flag is set in configuration file via the:
$ sentinelhub.config --use_s3_l1c_bucket true;

but when doing an actual tile request for that bucket l1c, the error access denied is given as as the configuration for access keys and secret keys has been setup right but the header x-amz-request-payer was not included. Going through download.py i noticed when querying bucket l2A, that header is already added in code.

 Is it possible to add the same header to the request of a l1c request as i think was done here.
download.py
_do_aws_request
   return s3_client.get_object(Bucket=bucket_name, Key=url_key, RequestPayer='requester') 
I don't think they use the same function as l1c throws Access Denied and l2A doesn't.

Sorry if i might be on the wrong track.

**Comment by AleksMat** [comment]
You are right, there is another problem:

In order to reduce the costs of download from AWS buckets some metadata files listed [here](https://forum.sentinel-hub.com/t/changes-of-the-access-rights-to-l1c-bucket-at-aws-public-datasets-requester-pays/172) are still available from http access. However the url address of these files has recently changed to [roda.sentinel-hub.com/sentinel-s2-l1c/](https://roda.sentinel-hub.com/sentinel-s2-l1c/). The package is still trying to access these data from old address and therefore gets `Access Denied`.

A quick fix that you can do at the moment is to change the address in config file by setting
```bash
sentinelhub.config --aws_base_url https://roda.sentinel-hub.com/sentinel-s2-l1c/
```
But we are going to fix this in the new version.

**Comment by AleksMat** [comment]
We have just released `sentinelhub` package version `2.4.1` with improvements regarding recent changes of L1C bucket access.

Now download from `s3` bucket in only option and metadata file are by default being downloaded from https://roda.sentinel-hub.com/sentinel-s2-l1c/ or https://roda.sentinel-hub.com/sentinel-s2-l2a/.

This should solve the issues.
